tests/CPT_test.py

```
@@ -0,0 +1,332 @@
+
+        def __len__(self):
+            return len(self.input_ids)
+
+        def __getitem__(self, idx):
+            return {
+                "input_ids": self.input_ids[idx],
+                "attention_mask": self.attention_mask[idx],
+                "input_type_mask": self.input_type_mask[idx],
+            }
+
+    dataset = CPTDataset(data, tokenizer, TEMPLATE)
+
+    return dataset
+
+
+def test_model_initialization_text(global_tokenizer, config_text):
+    """Test model loading and PEFT model initialization."""
+    base_model = AutoModelForCausalLM.from_pretrained(MODEL_NAME, cache_dir=".", trust_remote_code=True)
+
+    model = get_peft_model(base_model, config_text)
+    assert model is not None, "PEFT model initialization failed"
+
+
+def test_model_initialization_random(global_tokenizer, config_random):
+    """Test model loading and PEFT model initialization."""
+    base_model = AutoModelForCausalLM.from_pretrained(MODEL_NAME, cache_dir=".", trust_remote_code=True)
+
+    model = get_peft_model(base_model, config_random)
+    assert model is not None, "PEFT model initialization failed"
+
+
+def test_model_training_random(sst_data, global_tokenizer, collator, config_random):
+    """Perform a short training run to verify the model and data integration."""
+
+    base_model = AutoModelForCausalLM.from_pretrained(MODEL_NAME, cache_dir=".", trust_remote_code=True)
+    model = get_peft_model(base_model, config_random)
+    emb = model.prompt_encoder.default.embedding.weight.data.clone().detach()
+    training_args = TrainingArguments(
+        output_dir="./results",
+        per_device_train_batch_size=1,
+        num_train_epochs=2,
+        remove_unused_columns=False,
+        save_strategy="no",
+        logging_steps=1,
+    )
+
+    train_dataset = dataset(sst_data["train"], global_tokenizer)
+
+    trainer = Trainer(model=model, args=training_args, train_dataset=train_dataset, data_collator=collator)
+
+    try:
```

No need to catch the exception, just let it fail.